src/NServiceBus.Core/EndpointConfiguration.cs

```diff
@@ -53,7 +53,7 @@ public EndpointConfiguration(string endpointName)
         /// <summary>
         /// Access to the pipeline configuration.
         /// </summary>
-        public PipelineSettings Pipeline => PipelineComponent.PipelineSettings;
+        public PipelineSettings Pipeline { get; }
```

Comment:
I'd rather see `public PipelineSettings PipelineSettings { get; }` as the property represents pipeline settings, not the pipeline.

Reply:
We can't do breaking changes at this stage. I think we would go with an extension method to align with all other APIs so:

`endpointConfiguration.Pipeline(p=>p.Register(...))`

Thoughts on that API?

Reply:
Sorry, I didn't realize that. The suggested API looks interesting. You'd have to show me more :)